src/cluster/shard/shard.go

```diff
@@ -97,18 +97,23 @@ type shard struct {
 	id                uint32
 	redirectToShardID *uint32
 
-	state        State
-	sourceID     string
-	cutoverNanos int64
-	cutoffNanos  int64
+	state         State
+	sourceID      string
+	destinationID string
+	cutoverNanos  int64
+	cutoffNanos   int64
 }
 
 func (s *shard) ID() uint32                        { return s.id }
 func (s *shard) State() State                      { return s.state }
 func (s *shard) SetState(state State) Shard        { s.state = state; return s }
 func (s *shard) SourceID() string                  { return s.sourceID }
+func (s *shard) DestinationID() string             { return s.destinationID }
```

Comment:
currently this type has a 1<>1 mapping with what's in the proto; i.e. we store in etcd. 

so i don't think it makes sense to add DestinationID to this type. This field isn't set in the proto to start with; and further, i'm guessing you're inferring it during the write path in some way - so keeping it localized to that package is better.

Reply:
just because it's needed in the dbnode package doesn't mean we have to tack it on here. for e.g. 
- why can't there be a different type for it? 
- or why keep it like this and explicitly maintain this extra state in etcd?